test/endtoend/transloadit/test.js

```diff
@@ -0,0 +1,42 @@
+/* global browser, expect, capabilities, $ */
+const path = require('path')
+const { selectFakeFile, supportsChooseFile } = require('../utils')
+
+const testURL = 'http://localhost:4567/transloadit'
+
+function unhideTheInput () {
+  var input = document.querySelector('#uppy-transloadit .uppy-Dashboard-input')
+  input.removeAttribute('hidden')
+  input.removeAttribute('aria-hidden')
+  input.removeAttribute('tabindex')
+}
+
+function setTransloaditKeyAndInit (transloaditKey) {
+  window.initUppyTransloadit(transloaditKey)
+}
+
+describe('Transloadit file processing', () => {
+  beforeEach(() => {
+    browser.url(testURL)
+    browser.execute(setTransloaditKeyAndInit, process.env.TRANSLOADIT_KEY)
+  })
+
+  it('should upload a file to Transloadit and crop it', () => {
+    const inputPath = '#uppy-transloadit .uppy-Dashboard-input'
+    const resultPath = '#uppy-result'
+
+    $(inputPath).waitForExist()
+
+    if (supportsChooseFile(capabilities)) {
+      browser.execute(unhideTheInput)
+      browser.chooseFile(inputPath, path.join(__dirname, '../../resources/image.jpg'))
+    } else {
+      browser.execute(selectFakeFile, 'uppyTransloadit')
+    }
+    browser.pause(15000)
+    // $('.uppy-StatusBar-actionBtn--upload').click()
+    // $(resultPath).waitForExist()
```

Comment:
it would be nice to avoid the hardcoded time here…did `waitForExist()` not work?

If that fails we could use `executeAsync` too:

```js
browser.executeAsync(function (done) {
  window.onResultAvailable = done
})
```

And then call `window.onResultAvailable()` from inside main.js.

Reply:
Yeah, it didn’t work, cause I didn’t realize it has timeout option, duh. Thanks, seems to work now.